core/store/mod.go

```diff
@@ -18,3 +18,9 @@ type Snapshot interface {
 	Readable
 	Writable
 }
+
+// Transaction is a generic interface that store implementations can use to
+// provide atomicity.
```

Comment:
I would also say that it "adds a handler function to be executed after the transaction successfully commits."